packages/proto-signing/src/signer.ts

```diff
@@ -0,0 +1,26 @@
+import { AccountData, OfflineSigner as OfflineAminoSigner, StdSignature } from "@cosmjs/launchpad";
+
+import { cosmos } from "./codec";
+
+export interface DirectSignResponse {
+  /**
+   * The sign doc that was signed.
+   * This may be different from the input signDoc when the signer modifies it as part of the signing process.
+   */
+  readonly signed: cosmos.tx.v1beta1.ISignDoc;
+  readonly signature: StdSignature;
+}
+
+export interface OfflineDirectSigner {
+  readonly getAccounts: () => Promise<readonly AccountData[]>;
+  readonly signDirect: (
+    signerAddress: string,
+    signDoc: cosmos.tx.v1beta1.ISignDoc,
+  ) => Promise<DirectSignResponse>;
+}
+
+export type OfflineSigner = OfflineAminoSigner | OfflineDirectSigner;
```

Comment:
Is this or exclusive? Can we have an `OfflineSigner` that implements both interfaces?

Reply:
It's the union, so you can implement both but you don't have to. E.g. this compiles:
```ts
export class TestSigner implements OfflineAminoSigner, OfflineDirectSigner {
  public constructor() {}

  public getAccounts(): Promise<readonly AccountData[]> {
    throw new Error("not implemented");
  }

  public signDirect(): Promise<DirectSignResponse> {
    throw new Error("not implemented");
  }

  public signAmino(): Promise<AminoSignResponse> {
    throw new Error("not implemented");
  }
}
```

Reply:
Good to know, thanks